### DISCUSSION TITLE
DIA-NN 1.9

### DISCUSSION THREAD

**Discussion by vdemichev** [vdemichev/DiaNN#1034] [General]
### DIA-NN 1.9 release summary
DIA-NN 1.9 is the biggest improvement of DIA-NN so far. Below is the summary of key features, please see the [documentation](https://github.com/vdemichev/DiaNN/blob/master/README.md) for details.

**Peptidoforms**
Data-dependent acquisition (DDA) has so far maintained one key advantage over data-independent acquisition (DIA): confidence in peptidoform assignment. That is, with DDA one can be reasonably confident that a peptide is matched to the spectrum of the correct peptidoform (i.e. without amino acid substitutions or other modifications), and that the set of reported modifications (phosphorylation, etc) is correct. Now we achieve this also with DIA, while maintaining all the advantages of DIA and largely preserving its deep proteome coverage. We expect a range of applications, from pQTL analysis in population proteomics to metaproteomics. A preprint describing the new peptidoform-scoring module in DIA-NN is to follow.

**Phosphoproteomics**
We use the new peptidoform scoring module to significantly improve phosphoproteomics workflows. Moreover, DIA-NN now reports site-specific localisation confidence along with site-level quantities, in a convenient format, greatly simplifying its use for phosphoproteomics.

**Multiplexing**
DIA-NN 1.9 features a second-generation [plexDIA](https://www.nature.com/articles/s41587-022-01389-w) (multiplexing) module, with a significantly enhanced ability to gain channel-specific confidence in peptide and protein identifications. Further, processing of multiplexed DIA data is greatly simplified by convenient output, including channel-specific protein group quantities obtained with QuantUMS.

**timsTOF proteomics**
DIA-NN 1.9 implements Slice-PASEF as well as features preliminary support for midia-PASEF and Synchro-PASEF.

**Quantification**
DIA-NN 1.9 features a second-generation [QuantUMS](https://doi.org/10.1101/2023.06.20.545604) module, wherein quantities are optimised with machine learning and statistically-justified accuracy estimates are available for individual quantities.

**Visualisation**
This has been the most often requested feature since the conception of DIA-NN. Now supported via either Skyline integration or via a dedicated DIA-NN Viewer.

**General performance**
Better identification numbers and stricter control of false discoveries, along with extensive options to tailor the identification and quantification confidence control to a specific experiment.

**Speed and code quality**
DIA-NN has been overhauled to match the modern coding practices using C++20, with a focus on efficient memory use and better multithreading. DIA-NN 1.9 features code optimisations which yield roughly 1.3x-2x speed gains for library-free search. Large predicted libraries (tens of millions of precursors) are now often 10x+ quicker to generate.

**Timeline**. This is a Windows release of DIA-NN 1.9, Linux support is to follow shortly. Further, we have a number of features and performance improvements under active development and will likely release a series of updates implementing these in the near future. We will also be grateful for any feedback on DIA-NN 1.9 as well as feature requests, which we will do our best to implement.

### Future roadmap
DIA-NN is under active development, towards (i) enabling new technologies as well as (ii) achieving better performance for existing workflows. In the latter case, we have the following planned or under development:

* While DIA-NN performs remarkably well in library-free setting already, there is a room for even better performance. Specifically, DIA-NN will in the future implement experiment-specific transfer learning, similar to the concept recently introduced in AlphaDIA.

* DIA-NN already implements a low RAM usage mode, which restricts the amount of system memory it needs for its search. Currently, the biggest factor in RAM usage by DIA-NN in the lib-free mode is the storage of the predicted library in memory, especially when using multiplexing. DIA-NN will in the future implement a different format for internal library storage, with fold-change lower memory requirements.

* The ultra-fast mode in DIA-NN is great for preliminary analyses (up to 5x faster), although it does sacrifice identification performance, as it implements a spectrum centric-like search strategy, which is inherently less sensitive. We have a different fast search mode in works, which will have minimum performance trade offs.

* We have a number of algorithms in works, which will fully explore the potential (in terms of both identification and quantification performance) of Slice-PASEF, midia- and Synchro-PASEF, Scanning SWATH and Orbitrap Astral. While the current algorithms perform remarkably well already, showing the potential of these technologies, we work on specific improvements that will further boost the performance.

* DIA-NN will in the future incorporate a module for detailed QC analysis of DIA runs.

* Together with our collaborators, we are developing some exciting new workflows combining different tools.

<hr /><em>This discussion was created from the release <a href='https://github.com/vdemichev/DiaNN/releases/tag/1.9'>DIA-NN 1.9</a>.</em>

**Comment by Soulaimane-Aboulouard** [comment]
Hi Vadim,

Would it be possible to add functionality to manage the minimum number of peptides identified per protein ? directly on the interface or by a line command ? (For example, we would like the software to be able to keep proteins having at least 2 peptides, including 1 unique.)

In addition, in the "report.pg_matrix" results file, it would be interesting to add a column indicating the number of peptides identified per protein, in order to allow filtering based on this criterion.

Best,
Soulaimane

**Reply by vdemichev** [reply]
Hi Soulaimane,

In general, yes. You mean in each run specifically? Or total number of peptides in an experiment? The latter is easy when using MBR, just filter the .tsv library to only include proteins with >= 2 peptides.

Yes, agree about pg_matrix. I think this is actually quite easy, will look into this.

Best,
Vadim

**Reply by Soulaimane-Aboulouard** [reply]
Hi Vadim,
Thank you

I mean if it's possible to determine how many peptides are identified per protein groups and if it's possible to set a minimum number of peptides from the DIANN interface when launching the query (this is the equivalent of the "Min. razor + unique peptides" functionality of the MaxQuant software).

In addition, it would be interesting to add the number of peptides identified per protein group in the report.pg_matrix result file, for example by adding a specific command line for those who wish to add this information

This feature would be very handy for many people who want to adjust the stringency criteria of protein groups.

Thank you !!

Soulaimane

**Reply by vdemichev** [reply]
Hi Soulaimane, 

We can add this number to pg_matrix.
About min number of peptides, to clarify, would you also like to have it run-specific (i.e. protein group can have 5 peptides matched total, but in some runs it's just one, and then the group is simply not reported in those runs)? This will affect data completeness significantly.

Best,
Vadim

**Reply by Soulaimane-Aboulouard** [reply]
Hi Vadim,

As for adding this number to "pg_matrix", I'm fine with that.

As for the minimum number of peptides, I would like this to be based on the total number of peptides identified per protein group, without specificity to any particular run.

Please confirm if this is clear and feasible.

(Ideally, it would also be great to have a new option allowing the inclusion of an additional parameter when querying the runs on the Library/database with the DIANN software.. This parameter would be equivalent to the "Min razor + unique peptide" option in MaxQuant, where you can specify the minimum number of peptides per protein per run. For example, if we set the "Min razor + unique peptide" option to 2, a protein with 5 peptides identified in Run A is retained in the pg_matrix file with a quantification/intensity value, but the same protein with 1 peptide identified in Run B is not represented in the pg_matrix and gets a NaN value.)

Sincerely,
Soulaimane

**Reply by longchung90** [reply]
Hi Soulaimane,

It's so great that you ask the question on the number of psm(s). This is quite a crucial part in downstream analysis I think as DEqMS package requires requires this parameter to normalise. 

Can I please ask if we can extract the psm table from DIANN? 

Thank you

**Reply by vdemichev** [reply]
Hi Soulaimane,

Yes, we can add total number of peptides in pg_matrix. I will not promise this makes into 1.9.2, but this is in the todo list and will be there soon.

Best,
Vadim

**Reply by vdemichev** [reply]
Hi Long,

What do you mean by a PSM table? That is, anything in addition to the pr_matrix and the main report produced by DIA-NN?

Best,
Vadim

**Reply by longchung90** [reply]
Hi vdemichev,

Thank you for your reply. I think it is similar to the number of peptides used to identify the protein. I am using DEqMS to normalise the MS data and it does require the number of peptides. Hence, in this case with DIA-NN, it would be great to have the number of peptides reported as well in the future version.

**Comment by pieeye75** [comment]
This is great, but as we saw with the previous beta version, numerous vendors are flagging virus/trojan issues with the file. Our IT will not let us install (fair enough too). I get this report when I run the exe through VirusTotal;
![image](https://github.com/vdemichev/DiaNN/assets/18672293/9107d28e-cc09-44d2-be59-1fb002b2833e)

Can you comment?

**Reply by vdemichev** [reply]
The system itself as well as the installer are checked by built-in Windows antivirus with cloud protection enabled.
If you test against enough antivirus tools, some will produce false positives simply by chance, because each of those tools is known to produce false positives. I suspect, mostly because their 'AI' algorithms get confused about multiple layers of packaging in the DIA-NN installer (i.e. installer itself, then the PDF plotter is a Python instance packaged into .exe, while diann.exe embeds deep learning models into its executable, for example). If there were a true positive there, why would other tools not detect it? Below is another part of VirusTotal analysis:

![image](https://github.com/vdemichev/DiaNN/assets/30595084/7d947ae0-0c65-4482-9670-3d16f2832b21)

Here's Kaspersky's dynamic analysis (it runs the installer):
![image](https://github.com/vdemichev/DiaNN/assets/30595084/a8ed50c1-e280-4de5-987c-fbdeb7b8a0fa)

**Reply by mziehm** [reply]
We have similar problems as @pieeye75. While I agree with @vdemichev statement that more tests mean more false positives, there rate of 11 of 67 is unusually high for virustotal, which is a standard way to check software. Having uploaded/checked many software programs there, seeing 1 or 2 false positives occurs relatively frequently, but not this extend in my experience.
It would be great, if you could find the reason for these detections, and best case resolve them.
The full "current" virustotal results can be found here https://www.virustotal.com/gui/file/abf5472f07e028e5fb8faa1747cfaf6a77bf87354f9e2c84efb3567f6a0fa59d?nocache=1

**Reply by vdemichev** [reply]
As I mentioned, we did a lot more in depth checks than the 2-second scan virustotal does. We obviously cannot control what third-party antivirus tools' heuristic 'AI' algorithms report. Their business model is to provide 'free scanning' options, show that they detect some threats and this way entice the user to install & pay. So low false positive rate is not exactly their priority. Now, I am not an expert in AI algorithms of antivirus software, but I can speculate that deep packaging of stuff in DIA-NN distribution can contribute to those AI algorithms getting suspicious.

Since this still creates a concern, today or tomorrow I will add a regular '.zip' folder, without installer / uninstaller (if you check individual files, the uninstaller produces most hits on virustotal), so anyone will be able to just extract the contents to a preferred location on their PC, and these will not be producing any virustotal hits.

**Reply by vdemichev** [reply]
We are using a tool called InstallForge to generate Setup files (which package all the DIA-NN files and allow one-click installation, as opposed to unpacking a .zip file). It's apparently known to create install files that get flagged by VirusTotal https://installforge.net/forums/viewtopic.php?p=1882. At the same time, it's a popular tool with a large number of users https://installforge.net/forums/viewforum.php?f=5. Given that no one ever reported a concrete confirmed identification of anything malicious there (only AI-based suspicions, apparently), I am inclined to think everything is fine with that tool. But for the peace of mind,  as mentioned above, I will add DIA-NN just as a .zip folder.

**Reply by vdemichev** [reply]
Ok, I have added DIA-NN.1.9.binaries.zip to the release (i.e. just unpacked DIA-NN files without using InstallForge). Virustotal does not report anything for it.

**Reply by mziehm** [reply]
Many thanks this is great!

**Reply by pieeye75** [reply]
Thank you for this, much appreciated. I never suspected it had a virus, but our IT department was adamant. They have to assume the one hit is true. Clearly that would have been a "no win" for all concerned.

**Comment by animesh** [comment]
Awesome @vdemichev 💯 Any update on Linux version/rpm maybe?

**Reply by vdemichev** [reply]
Thanks! Will work on this hopefully this weekend. How fast we get it ready - depends on whether all goes smooth with dependencies. We basically need to overhaul the whole build system for the new version.

**Reply by vdemichev** [reply]
But on Windows it works fine with Clang, which we use for debug purposes, so hopefully will be smooth on Linux too :)

**Reply by animesh** [reply]
Thanks @vdemichev looking forwards to it 💯 On windows i noticed that running on two separate machines (
[machine.1.log.txt](https://github.com/user-attachments/files/15897988/machine.1.log.txt)
[machine.2.log.txt](https://github.com/user-attachments/files/15898381/machine.2.log.txt)) gave slightly different results ![image](https://github.com/vdemichev/DiaNN/assets/21457/91e157f2-319b-4104-90b8-e2e2ba0200c5)

scatter of values from respective *report.pg_matrix.tsv* with R^2 in blue are shown below

![image](https://github.com/vdemichev/DiaNN/assets/21457/f01b5e41-b1c5-4013-919d-4c65e250af4d)

and its log2-transformed values

![image](https://github.com/vdemichev/DiaNN/assets/21457/27162663-03fb-4fa7-a046-fd43b1a65d48)
is this expected?

**Reply by vdemichev** [reply]
Quant is different primarily because of the contaminants FASTA not loading for some reason on one of the PCs.

About IDs, was the predicted library the same, i.e. was generated on one of the PCs and then transferred to another PC?
Do you think you could share F:\promec\TIMSTOF\LARS\2024\240222_DIA\DIA_test\DIANN\mouse.tsv from the first machine and C:\Users\animeshs\libDIA\DIANN\mouse.tsv from the second?

**Reply by animesh** [reply]
Yes, the library is copied from one to another 🤞Loaded up the Machine 1 tsv https://drive.google.com/file/d/1noPOcf8Sw0K4BzbYP_FhIFJU3EyiGcX-/view?usp=sharing and 2 https://studntnu-my.sharepoint.com/:u:/g/personal/animeshs_ntnu_no/EXTNwIfCHQFOtcga6FYTNxYByT-bIbVnXWAqtw2bKxawyw?e=ldXyUO 👍

BTW how to make sure that contaminants are loaded properly?

**Reply by vdemichev** [reply]
Thanks for sharing! Will take a look.
Contaminants - if you are launching diann from the command line, navigating first to the DIA-NN directory should work.

**Reply by vdemichev** [reply]
OK, when DIA-NN needs to save any real number into a .tsv file, it is always rounded to the precision of 8 digits. Apparently, what happened here the rounding was different in the two runs. That is, the same fragment mass was saved in one case as 592.20312 and in another case as 592.20313. This number is not even undergoing any calculations in DIA-NN, it's directly copied form .predicted.speclib to memory and from memory written to .tsv. Same issues with RTs, although those are calculated. So for whatever reason (this remains to be investigated), these here are printed a bit differently on different machines by the same code.

My best guess so far is this: when you launch DIA-NN from the command line, the 'current working directory' is different from the DIA-NN folder. The DLLs containing the standard C++ library functions, including printing real numbers to a file, are now not embedded in DIA-NN but distributed with it. But Windows has a particular DLL search order, and I am not sure what it is in this case, but it might be that a system DLL is being used instead of the packaged one. Or it might be that the system DLL is always used, and these are different for these machines, leading to output discrepancy. 

I would expect:
- The output would be different always whenever you analyse any runs (with enough IDs to see a difference) with MBR. That is, can just analyse quick two lightweight ones with any spectral library - will see this difference in output for the second pass of MBR between the machines.
- It might be that the difference only manifests when DIA-NN is launched from command line, from outside of its installation folder. I.e. output is identical if using the GUI in each case.
- If it manifests always, regardless of how it is launched, then this is due to loading system DLLs.

**Reply by animesh** [reply]
Thanks @vdemichev for looking into this 🙏  will launch it from the directory and check again 🤞 BTW how far are we from Linux release 🤓

**Reply by vdemichev** [reply]
In the process :))

**Reply by vdemichev** [reply]
"will launch it from the directory " - will take a lot of time with this experiment. But if you could try to repeat the same approach on something which is quick to analyse (any empirical lib + two files), I would be grateful.

**Reply by animesh** [reply]
Tried two files via command line from withing the install exe dir, logs 
[mouse-report.log.txt](https://github.com/user-attachments/files/15910695/mouse-report.log.txt)
 
[mouse-report.log.txt](https://github.com/user-attachments/files/15910710/mouse-report.log.txt)

and correlation is perfect 👍🏽 

![image](https://github.com/vdemichev/DiaNN/assets/21457/d1506875-b98d-4a6b-9822-91657fbccdb5)

uploading those *tsv* files 

Machine 1 https://drive.google.com/file/d/1oBuMR3VExK-9tb0jWq8q9BxHRUjoOLPM/view?usp=sharing , 
Machine 2 https://drive.google.com/file/d/1o_pzTbG1D-09CZIgcegGcxwupugKhQQl/view?usp=sharing respectively, are there still some differences with ?

**Reply by vdemichev** [reply]
Thank you!
I will need to think what caused the first inconsistency.
Also, we will probably switch to .parquet format for DIA-based spectral libraries (i.e. still support .tsv as input, but new libs will be generated as .parquet), and it will then eliminate any kind of issues like this in the future, as it is a precise binary format.

**Comment by kometkamerat** [comment]
Hi! Seems like the program refuses to run or install on Windows Server 2019. Version 1.8.1 runs without problems. 

Error message from installer:
<img width="393" alt="image" src="https://github.com/vdemichev/DiaNN/assets/38461720/2753e5b1-b875-4235-9403-5badbd36315a">

Unpacking the biaries and starting the executables just results in nothing. No program starts and no error message in event log. 

K.

**Reply by vdemichev** [reply]
If you unpack the .binaries.zip, navigate to the respective folder in command prompt, and launch diann.exe from the command prompt, what does it print?

**Reply by kometkamerat** [reply]
Interresting, so running diann.exe or DIA-NN.exe from Powershell works perfectly, just not double-clicking from the File Explorer... Creating a desktop shortcut does not work either. But I'll direct my users to run it from powershell for now.

**Reply by vdemichev** [reply]
Then a .bat file launching DIA-NN.exe should also work I guess... Strange indeed

**Comment by xiaosiruwei** [comment]
There is no diann.cpp file in the source code of versions 1.8.1 and 1.9. Is it not open-source？

**Reply by vdemichev** [reply]
No, not open source

**Comment by vdemichev** [comment]
A quick update about the Linux version, I have managed to build DIA-NN under Linux with all the dependencies now, i.e. all the technical difficulties have been solved. So what's left is testing, plus I would like to incorporate some new functionality (have received several great suggestions) - and then will release 1.9.1 for both Windows and Linux.

**Reply by Aaron-Amunix** [reply]
Great, thanks a lot for all your hard work on DIA-NN.  I tested DIA-NN 1.9.1 on my unique substrate library data and it works quite well.  We currently are using DIA-NN 1.8 in a Linux environment and am excited to try 1.9.1 soon!  Thanks again for making such an awesome tool and keeping on building.  Hopefully I'll see you again at HUPO this year!

**Comment by vmataf** [comment]
Hi Vadim,
I run a dataset with DIANN 1.8.1 and DIANN 1.9.1 and I compared results.
Number of ProteinGroups identified were similar : 8261 for DIANN1.8.1 and 8134 for DIANN1.9.1
Number of Peptides  identified were similar: 94767 for DIANN1.8.1 and 105210 for DIANN 1.9.1
however the coefficient of variation was different: medium CV 0.12 for DIANN1.8.1 while 0.35 for DIANN1.9.1 with bimodal shape
[CV.pdf](https://github.com/user-attachments/files/17202725/CV.pdf)
This reflects on statistically significant proteins that were low for DIANN1.9.1.
I used the default setting for DIANN 1.9.1
 -(direct DIA fasta digest for library free enabled)  
-no peptidoform
 -MBR , QUantUMS high precision and RT dependent normalization enambled.
Do you have any suggestion why DIANN1.9.1 quantitation has higher variability?

**Reply by vdemichev** [reply]
Hi, 

Can you please share the (i) log of 1.9.1 and (ii) pg_matrix output of both 1.9.1 and 1.8.1?

> (direct DIA fasta digest for library free enabled)

DIA-NN 1.9.1 prints a warning here that 'on the fly digest + analysis' must not be used. If that was the case, this likely explains weird CVs.

Best,
Vadim

**Comment by vmataf** [comment]
Hi,

here the files you requested.
Thanks a lot
Vittoria

--

Vittoria Matafora
IFOM | Staff Scientist | Proteomics Unit
Ph.: +39 02 57430 3302 | ***@***.******@***.***>
www.ifom.eu<https://www.ifom.eu> | Linkedin<https://www.linkedin.com/school/ifom/> | Instagram<https://www.instagram.com/ifom_milan/?hl=it> | Facebook<https://www.facebook.com/ifom.eu> | Twitter<https://twitter.com/ifomresearch> | YouTube<https://www.youtube.com/channel/UC7J0QYrvQH87ZhWKJshQVgw>
IFOM ETS - The AIRC Institute of Molecular Oncology | Via Adamello 16, 20139 Milan, Italy

[cid:logoIfom_2d2161e0-3259-466c-8931-26f8c2a5979f.png]

________________________________

Confidentiality notice. This message, together with its annexes, contains information to be deemed strictly confidential and is destined only to the addressee(s) identified above who only may use, copy and, under his/their responsibility, further disseminate it. if anyone received this message by mistake or reads it without entitlement is forewarned that keeping, copying, disseminating or distributing this message to persons other than the addressee(s) is strictly forbidden and is asked to transmit it immediately to the sender and to erase the original message received.
Thank you
Save a tree - Do not print this email unless absolutely necessary

________________________________

**Comment by vdemichev** [comment]
Likely this:

> WARNING: it is strongly recommended to first generate an in silico-predicted library in a separate pipeline step and then use it to process the raw data, now without activating FASTA digest

**Comment by vmataf** [comment]
Hi Vadim,

I run the analysis as you suggested,

-first I run  in silico predicted library from fasta,  with fast digest for library free enabled (without loading .raw files)

- then I run the analysis with .raw files, loading in silico predicted spectral library and  with fast digest for library free not enabled.

I attached log file. Again CV showed the same shape.

Might you help me again?

Thanks

Vittoria




________________________________

**Comment by vmataf** [comment]
Hi,

 I have just realized that DIANN does not support FIAMS with mulyiple CVs

-FAIMS with multiple CVs is supported after splitting the runs,

All my samples have multiple CVs, I did not split the runs ...could this affect the coefficient of variation?

Thanks



________________________________

**Reply by vdemichev** [reply]
Yes, this would explain then I guess...

**Comment by vmataf** [comment]
Hi Vadim,

I splitted the FAIMS CV for all the .raw files , I run all the CV as separate experiments in DIANN 1.9.1, I increased the number of proteingroups identified, however I got the same coefficient of variation CV shape.

I do not understand why, in version 1.8.1 I do not have same problem.


--

Vittoria Matafora
IFOM | Staff Scientist | Proteomics Unit
Ph.: +39 02 57430 3302 | ***@***.******@***.***>
www.ifom.eu<https://www.ifom.eu> | Linkedin<https://www.linkedin.com/school/ifom/> | Instagram<https://www.instagram.com/ifom_milan/?hl=it> | Facebook<https://www.facebook.com/ifom.eu> | Twitter<https://twitter.com/ifomresearch> | YouTube<https://www.youtube.com/channel/UC7J0QYrvQH87ZhWKJshQVgw>
IFOM ETS - The AIRC Institute of Molecular Oncology | Via Adamello 16, 20139 Milan, Italy

[cid:logoIfom_2d2161e0-3259-466c-8931-26f8c2a5979f.png]

________________________________

Confidentiality notice. This message, together with its annexes, contains information to be deemed strictly confidential and is destined only to the addressee(s) identified above who only may use, copy and, under his/their responsibility, further disseminate it. if anyone received this message by mistake or reads it without entitlement is forewarned that keeping, copying, disseminating or distributing this message to persons other than the addressee(s) is strictly forbidden and is asked to transmit it immediately to the sender and to erase the original message received.
Thank you
Save a tree - Do not print this email unless absolutely necessary

________________________________

**Reply by vdemichev** [reply]
>  I run all the CV as separate experiments in DIANN 1.9.1

So are coefficients of variation fine within each such experiment?

**Comment by vmataf** [comment]
The coefficients of variation within each such experiment look always wired,
separate and/or together look the same


--

Vittoria Matafora
IFOM | Staff Scientist | Proteomics Unit
Ph.: +39 02 57430 3302 | ***@***.******@***.***>
www.ifom.eu<https://www.ifom.eu> | Linkedin<https://www.linkedin.com/school/ifom/> | Instagram<https://www.instagram.com/ifom_milan/?hl=it> | Facebook<https://www.facebook.com/ifom.eu> | Twitter<https://twitter.com/ifomresearch> | YouTube<https://www.youtube.com/channel/UC7J0QYrvQH87ZhWKJshQVgw>
IFOM ETS - The AIRC Institute of Molecular Oncology | Via Adamello 16, 20139 Milan, Italy

[cid:logoIfom_2d2161e0-3259-466c-8931-26f8c2a5979f.png]

________________________________

Confidentiality notice. This message, together with its annexes, contains information to be deemed strictly confidential and is destined only to the addressee(s) identified above who only may use, copy and, under his/their responsibility, further disseminate it. if anyone received this message by mistake or reads it without entitlement is forewarned that keeping, copying, disseminating or distributing this message to persons other than the addressee(s) is strictly forbidden and is asked to transmit it immediately to the sender and to erase the original message received.
Thank you
Save a tree - Do not print this email unless absolutely necessary
From: Vadim Demichev ***@***.***>
Reply-To: vdemichev/DiaNN ***@***.***>
Date: Thursday, October 3, 2024 at 12:47 PM
To: vdemichev/DiaNN ***@***.***>
Cc: Vittoria Matafora ***@***.***>, Comment ***@***.***>
Subject: Re: [vdemichev/DiaNN] DIA-NN 1.9 (Discussion #1034)


I run all the CV as separate experiments in DIANN 1.9.1

So are coefficients of variation fine within each such experiment?

—
Reply to this email directly, view it on GitHub<https://urlsand.esvalabs.com/?u=https%3A%2F%2Fgithub.com%2Fvdemichev%2FDiaNN%2Fdiscussions%2F1034%23discussioncomment-10830681&e=c326a8c7&h=8509831a&f=y&p=y>, or unsubscribe<https://urlsand.esvalabs.com/?u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FBLX3TQLQH4VWRPGAV34SSUTZZUOFPAVCNFSM6AAAAABJBC5L66VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOBTGA3DQMI&e=c326a8c7&h=77db6e78&f=y&p=y>.
You are receiving this because you commented.Message ID: ***@***.***>

--
This message has been checked by Libraesva ESG and is found to be clean.
Report as bad/spam<https://esva.ifom.eu/action/4XK7gF3zSVzLvNF/report-as-bad>
Blocklist sender<https://esva.ifom.eu/action/4XK7gF3zSVzLvNF/blocklist>

**Comment by vmataf** [comment]
Hi Vadim,

I fixed the problem of coefficient of variation, now it is fine. I run all the CV as separate experiments in DIANN 1.9.1.
Next step is to combine the CV channels. You suggested to

•  combine the reports using R, select the 'best' channel for each precursor ion, discard all other channels for this precursor

On which criteria do you select the 'best' channel for each precursor ion?

Moreover

•  perform protein quantification using MaxLFQ with the diann R package or the "iq" R package (the latter is currently much faster)

How to do that?

Thanks a lot

Vittoria





--

Vittoria Matafora
IFOM | Staff Scientist | Proteomics Unit
Ph.: +39 02 57430 3302 | ***@***.******@***.***>
www.ifom.eu<https://www.ifom.eu> | Linkedin<https://www.linkedin.com/school/ifom/> | Instagram<https://www.instagram.com/ifom_milan/?hl=it> | Facebook<https://www.facebook.com/ifom.eu> | Twitter<https://twitter.com/ifomresearch> | YouTube<https://www.youtube.com/channel/UC7J0QYrvQH87ZhWKJshQVgw>
IFOM ETS - The AIRC Institute of Molecular Oncology | Via Adamello 16, 20139 Milan, Italy

[cid:logoIfom_2d2161e0-3259-466c-8931-26f8c2a5979f.png]

________________________________

Confidentiality notice. This message, together with its annexes, contains information to be deemed strictly confidential and is destined only to the addressee(s) identified above who only may use, copy and, under his/their responsibility, further disseminate it. if anyone received this message by mistake or reads it without entitlement is forewarned that keeping, copying, disseminating or distributing this message to persons other than the addressee(s) is strictly forbidden and is asked to transmit it immediately to the sender and to erase the original message received.
Thank you
Save a tree - Do not print this email unless absolutely necessary
From: Vadim Demichev ***@***.***>
Reply-To: vdemichev/DiaNN ***@***.***>
Date: Thursday, October 3, 2024 at 12:47 PM
To: vdemichev/DiaNN ***@***.***>
Cc: Vittoria Matafora ***@***.***>, Comment ***@***.***>
Subject: Re: [vdemichev/DiaNN] DIA-NN 1.9 (Discussion #1034)


I run all the CV as separate experiments in DIANN 1.9.1

So are coefficients of variation fine within each such experiment?

—
Reply to this email directly, view it on GitHub<https://urlsand.esvalabs.com/?u=https%3A%2F%2Fgithub.com%2Fvdemichev%2FDiaNN%2Fdiscussions%2F1034%23discussioncomment-10830681&e=c326a8c7&h=8509831a&f=y&p=y>, or unsubscribe<https://urlsand.esvalabs.com/?u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FBLX3TQLQH4VWRPGAV34SSUTZZUOFPAVCNFSM6AAAAABJBC5L66VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOBTGA3DQMI&e=c326a8c7&h=77db6e78&f=y&p=y>.
You are receiving this because you commented.Message ID: ***@***.***>

--
This message has been checked by Libraesva ESG and is found to be clean.
Report as bad/spam<https://esva.ifom.eu/action/4XK7gF3zSVzLvNF/report-as-bad>
Blocklist sender<https://esva.ifom.eu/action/4XK7gF3zSVzLvNF/blocklist>

**Reply by vdemichev** [reply]
Hi Vittoria,

> On which criteria do you select the 'best' channel for each precursor ion?

E.g. the lowest global q-value for the precursor across the experiments with different CVs. Or highest data completeness.

> How to do that?

Please see the respective docs for the iq R package, it's fairly simple but requires a bit of coding in R (like ~20-30 lines maybe).

Best,
Vadim

**Comment by dorvashmr** [comment]
Hi Vadim,

I was wondering if you could help me know what are the changes made from the 1.9.0 to 1.9.1 versions? Particularly if there are any FDR- or quantity calculation-related changes.

Cheers,
Reza

**Reply by dorvashmr** [reply]
Hi again Vadim,

My current observation with these two versions is:
The same dataset (mixed-species) was searched with the same settings using both versions (lib-free, FDR=1%, MBR+, Double-pass, QuantUMS high acc, Global normaization).

With 1.9.0 -> protein groups =  ~5000 ; experimentally calculated precursor FDR = ~0.5-0.7% ;  experimentally calculated protein FDR = ~2.5-3.5%
With 1.9.1 -> protein groups = ~11000 ; experimentally calculated precursor FDR = ~2-3% ;  experimentally calculated protein FDR = ~20%


I was wondering if you could help me find what is causing this observation.

Cheers,
Reza

**Reply by vdemichev** [reply]
Hi Reza,

I assume the in silico library prediction from FASTA and raw data analysis is combined in one step?

Best,
Vadim

**Reply by dorvashmr** [reply]
Hi Vadim,

With both versions, the FASTA was loaded with the raw (wiff.dia) files and the "FASTA digest ..." and "Deep learnin ..." options were selected. The search in both versions was double-pass, though.

Cheers,
Reza

**Reply by vdemichev** [reply]
Hi Reza,

In this case 1.9.1 behaves incorrectly (please note that it also prints a warning that settings are incorrect). 1.9.2 should work correctly (but still print a warning, so please create an in silico library in a separate pipeline step).

Best,
Vadim

**Comment by karlssoc** [comment]
DIA-NN has 130(!) available command line parameters. Maybe a collection of configuration files for different types of analyses could help users get started. For example: 
- TimsTOF libfree peptide 
- Astral phosphopeptides 
- Sciex Scanning swath
- ...

**Reply by vdemichev** [reply]
Hi,

It's 130, but for most experiments you need to use none :) 
All the common settings are configured through the GUI, and this takes just a minute or two, we describe the steps to configure here: https://github.com/vdemichev/DiaNN?tab=readme-ov-file#changing-default-settings.

Best,
Vadim
